### PR TITLE
docs - postgres_fdw improvements, new options

### PR DESCRIPTION
**Pull request by lisakowen**
docs for https://github.com/greenplum-db/gpdb/pull/15821, https://github.com/greenplum-db/gpdb/pull/15824, https://github.com/greenplum-db/gpdb/pull/15857: postgres_fdw enhancements.

in this PR:  update the postgres_fdw topic for new "distributed" support, including mentioning the greenplum and new postgres_fdw-specific CREATE SERVER options.  also mention new partial agg pushdown support.
  
questions:
1. does gporca support any aspect of distributed postgres_fdw?  looks like the partial agg pushdown tests are using planner only.
2. i used the test files in the PRs to get some idea about what is not supported.  not sure if i captured this info correctly, please advise.
3. are there any new limitations or features when using the fdw to access Greenplum to note?
 
doc review site link (behind vpn):  https://docs-staging.vmware.com/en/draft/VMware-Greenplum/postgres_fdw-improve/greenplum-database/GUID-ref_guide-modules-postgres_fdw.html

### CODE REVIEW

File: gpdb-doc/markdown/ref_guide/modules/postgres_fdw.html.md [hunk 1]
```diff
@@ -1,18 +1,51 @@
...
+| multi_hosts | Space-separated list of remote PostgreSQL server host names. | You must specify exactly `num_segments` number of hosts in the list. |
+| multi_ports | Space-separated list of port numbers for the PostgreSQL servers. | You must specify exactly one port number for each host specified in `multi_hosts`, in order. |
+
+Setting these options instructs `postgres_fdw` to treat a foreign table that you create that specifes this `SERVER` as a *distributed* foreign table. That is, a foreign table whose underlying data is stored on multiple remote PostgreSQL servers. `postgres_fdw` directs a query on such a foreign table to each PostgreSQL server specified in `multi_hosts`.
+
+## <a id="pushdown"></a>About Aggregate Pushdown Support
+
+`postgres_fdw` supports partial aggregate pushdown for a *distributed* query under the following conditions:
+
+- The aggregate contains no `DISTINCT`, `FILTER`, or `ORDER BY` clauses.
```
Comment (zhaorui20005) [inline gpdb-doc/markdown/ref_guide/modules/postgres_fdw.html.md:30]: I think `filter` and `group by` should be supported to be pushed down, except when there are unshipable expressions in the clauses.

**Comment by zhaorui20005** [comment]
>questions:
>1. does gporca support any aspect of distributed postgres_fdw?  looks like the partial agg pushdown tests are using planner only.

Yes, orca does not support push down, and when num_segments is larger then the default segments numbers, the orca planner will fall back.
>2. i used the test files in the PRs to get some idea about what is not supported.  not sure if i captured this info correctly, please advise.
 
I think filter and group by should be pushed down for normal expressions expect there are unshipable conditions in the clauses.
>3. are there any new limitations or features when using the fdw to access Greenplum to note?

What you have written down looks good to me.

**Comment by lisakowen** [comment]
@zhaorui20005 - thank you for reviewing.  i added a few notes for the GPORCA unsupported cases.

i also (removed the statement about GROUP BY and FILTER.  not sure what an unshipable conditions is - should we mention this?)

**Comment by zhaorui20005** [comment]
> @zhaorui20005 - thank you for reviewing. i added a few notes for the GPORCA unsupported cases.
> 
> i also (removed the statement about GROUP BY and FILTER. not sure what an unshipable conditions is - should we mention this?)

I don't think we need to mention unshipable conditions, they are just inner function for postgres_fdw, and they are not unique to this new feature, the legacy postgres_fdw cannot push down unshipable conditions(like unstable functions) to remote too.

File: gpdb-doc/markdown/ref_guide/modules/postgres_fdw.html.md [hunk 2]
```diff
@@ -1,18 +1,54 @@
...
+|-------------|-----|-------------|
+| mpp_execute | Greenplum Database option that identifies the host(s) from which `postgres_fdw` reads or writes data. | Set to `'all segments'`, which, when specified for `postgres_fdw`, translates to all remote PostgreSQL servers specified in `multi_hosts`. |
+| num_segments<sup>1</sup> | Greenplum Database option that identifies the number of query executors that Greenplum spawns on the source Greenplum cluster. | Set to the number of remote PostgreSQL servers. If this option is not set, defaults to the number of segments in the local Greenplum cluster.|
+| multi_hosts | Space-separated list of remote PostgreSQL server host names. | You must specify exactly `num_segments` number of hosts in the list. |
+| multi_ports | Space-separated list of port numbers for the PostgreSQL servers. | You must specify exactly one port number for each host specified in `multi_hosts`, in order. |
+
+<sup>1</sup> The Greenplum Query Optimizer (GPORCA) can not plan and optimize queries when `num_segments` is larger than the number of segments in the local Greenplum cluster; such queries always fall back to the Postgres Planner.
```
Comment (zhaorui20005) [inline gpdb-doc/markdown/ref_guide/modules/postgres_fdw.html.md:24]: Here is my last comments' fault. 
When `num_segments` is not equal to the GPDB cluster number of segments, GPORCA will fall back to postgres planner, not only larger then number of segments in the local Greenplum cluster.

File: gpdb-doc/markdown/ref_guide/modules/postgres_fdw.html.md [hunk 3]
```diff
@@ -1,18 +1,54 @@
...
+
+- The aggregate contains no `DISTINCT` or `ORDER BY` clauses.
+- The aggregate does not contain `HAVING` clause.
+- The aggregate function is not `array_agg()`.
+- The query contains no `LIMIT` or `JOIN` clauses.
+
+> **Note** `postgres_fdw` does not support partial aggregate pushdown when the Greenplum Query Optimizer (GPORCA) is enabled for a query.
+
+
+## <a id="topic_gp_limit"></a>Limitations
```
Comment (mperezfuster) [inline gpdb-doc/markdown/ref_guide/modules/postgres_fdw.html.md:40]: maybe rename to postgres_fdw Limitations or Foreign-data Wrapper Limitations?

**Review by mperezfuster — COMMENTED** [review]
Thanks for creating this page! I had a couple of comments.

**Review by mperezfuster — APPROVED** [review]

**Comment by lisakowen** [comment]
merging this PR.  i will address further comments in a new PR.

**Pull request merged**